gems/aws-sdk-iotdataplane/spec/client_spec.rb

```diff
@@ -28,17 +28,27 @@ module IoTDataPlane
       it 'correctly extracts the sigv4 signing region' do
         client = Client.new(
           region: "us-east-1",
-          endpoint: "https://FOOBARFOOBAR.iot.us-east-1.amazonaws.com"
+          endpoint: "https://FOOBARFOOBAR.iot.us-east-1.amazonaws.com",
+          stub_responses: true
         )
-        expect(client.config.sigv4_region).to eq("us-east-1")
+        expect(Aws::Plugins::Sign).to receive(:signer_for).and_wrap_original do |m, *args|
+          expect(args.first['signingRegion']).to eq('us-east-1')
+          m.call(*args)
+        end
+        client.get_thing_shadow(thing_name: 'thing')
       end
 
       it 'correctly extracts the sigv4 signing region outside of us-east-1' do
         client = Client.new(
           region: "eu-west-1",
-          endpoint: "https://FOOBARFOOBAR.iot.eu-west-1.amazonaws.com"
+          endpoint: "https://FOOBARFOOBAR.iot.eu-west-1.amazonaws.com",
+          stub_responses: true
         )
-        expect(client.config.sigv4_region).to eq("eu-west-1")
+        expect(Aws::Plugins::Sign).to receive(:signer_for).and_wrap_original do |m, *args|
```

Comment:
I think we can abstract this into spec helper and operationInput tests can also use it.